### PR TITLE
[TGUI] ImageButton component

### PR DESCRIPTION
**Pull request by AyIong**
## What Does This PR Do
New TGUI component - ImageButton!
Documentation is included, any suggestions will be appreciated

It was created to solve several problems:
1. To improve the performance of buttons with images
2. Improve the convenience of creating buttons with images
3. Standardise buttons with images

Replaces custom styled buttons in Prize Counter to ImageButton.
There aren't many differences in appearance, but it looks more polished in the little things.

<details>
<summary>How it looks like</summary>

| Before | After |
| - | - |
| image | image |

</details>

In addition, the render time when opening the window was reduced from 8-10 frames to 5-6 frames.
Not much, but its something.
<details>
<summary>Render time</summary>

| Before | After |
| - | - |
| image | image |

</details>

Also added 1 more color - gold.

## Why It's Good For The Game
Beautiful buttons that are easy to embed images into.
The component itself is not very heavy, unlike variants where the image is embedded in a regular Button component.
In the end, I tried to make the component as convenient as possible for developers

## Images of changes
You can see and test it into KitchenSink
image

## Testing
Yes, on the BayStation12 downstream SS220 with identical TGUI on 3 UI's 
And here on PrizeCounter

Anyway there are a couple of problems worth solving:
1. You can't overlay an icon on a image
2. ImageButton.Item is quite expensive and increases render time by 1.5-2 times.
3. ImageButton.Item has no vertical option
~~4. ImageButton.Item has no horizontal option to position the icon horizontally~~

They are not very critical and there is no way to fix them at this time (Classic TODO 10 years later)

## Changelog
:cl:
tweak: A little polished PrizeCounter UI
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/styles/colors.scss [hunk 1]
```diff
@@ -29,6 +29,8 @@ $average: #f08f11 !default;
 $bad: #db2828 !default;
 $label: #7e90a7 !default;
 
+$gold: #f2a918 !default;
```
Comment (larentoun) [inline tgui/packages/tgui/styles/colors.scss:32]: Probably move it closer to base colors, after yellow?

Comment (AyIong) [reply tgui/packages/tgui/styles/colors.scss:32]: Maybe. but the logic is:
Base colours => State colours => Custom colours

Comment (Burzah) [inline tgui/packages/tgui/styles/colors.scss:32]: I am fine with adding this to base colors. Though, if we want to start a custom color group, can you add a new line with a comment for that? Also while we are here, can we add another comment line to mark State colors?

Comment (AyIong) [reply tgui/packages/tgui/styles/colors.scss:32]: Good idea

**Comment by AyIong** [comment]
> This pull request seems to be stale as there have been no changes in 14 days, please make changes within 7 days or the PR will be closed. If you believe this is a mistake, please inform a development team member on Discord.

Listen here, you little shit
image

**Review by Burzah — APPROVED** [review]

**Review by S34NW — COMMENTED** [review]
Is this compliant with Edge?

**Comment by AyIong** [comment]
> Is this compliant with Edge?

I don't know xdd
Need testing on 516

**Review by S34NW — APPROVED** [review]
now with 100% more 516 compatability

**Pull request merged**